engine/src/main/java/org/terasology/rendering/nui/widgets/UIImageSlideshow.java

```diff
@@ -128,22 +116,73 @@ public void clean() {
         images = new ArrayList<>();
     }
 
-    protected boolean isActive() {
-        return active;
-    }
-
     /**
-     * Starts the slideshow.
+     * Starts automatic slideshow.
      */
     public void start() {
         active = true;
     }
 
     /**
-     * Stops the slideshow.
+     * Stops automatic slideshow.
      */
     public void stop() {
         active = false;
     }
 
+    /**
+     * Switches to next image of list.
+     */
+    public void nextImage() {
+        int size = images.size();
+        int prevIndex = index;
+        if (size > 0) {
+            if (index == size - 1) {
+                if (infinite) {
+                    index = 0;
+                } else {
+                    stop();
+                }
+            } else {
+                index++;
+            }
+            if (prevIndex != index) {
+                currentImage.set(images.get(index));
```

Comment:
### Option B
Change this from setting `currentImage` to the new image, just set the _texture_ to the new texture.

If you wanted to get really fancy you could also use a binding, and bind the texture to `images.get(index)` via `ReadOnlyBinding` this would make the texture automatically update without you having to set it.

Reply:
To be honest, I've understood that this kind of bindings decreases performance of whole UI. 
And in all my recent PRs I'm trying do not use it too often, only where it really might help.